pkg/tsdb/testdatasource/scenarios.go

```diff
@@ -285,42 +285,40 @@ func (p *testDataPlugin) handleManualEntryScenario(ctx context.Context, req *bac
 	for _, q := range req.Queries {
 		model, err := simplejson.NewJson(q.JSON)
 		if err != nil {
-			continue
+			return nil, fmt.Errorf("error reading query")
 		}
 		points := model.Get("points").MustArray()
 
 		frame := newSeriesForQuery(q, model, 0)
-		startTime := q.TimeRange.From.UnixNano() / int64(time.Millisecond)
-		endTime := q.TimeRange.To.UnixNano() / int64(time.Millisecond)
 
-		timeVec := make([]*time.Time, 0)
-		floatVec := make([]*float64, 0)
+		timeField := data.NewFieldFromFieldType(data.FieldTypeTime, 0)
+		valueField := data.NewFieldFromFieldType(data.FieldTypeNullableFloat64, 0)
```

Comment:
this change it not really necessary... but worth noting that creating fields from a slice will copy them an extra time vs this method that appends directly to the final vaue